src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java

```diff
@@ -1,151 +1,52 @@
-    @Shadow
-    private static float blue;
-    private static int waterFogColor = -1;
-    private static int nextWaterFogColor = -1;
-    private static long lastWaterFogColorUpdateTime = -1L;
-
-
-
-
     /**
-     * @author
-     * @reason
+     * @author AnOpenSauceDev
+     * @reason what fog??!?!? never heard of it.
      */
-    @Overwrite
-    public static void render(Camera camera, float tickDelta, ClientWorld world, int viewDistance, float skyDarkness) {
-
-            //Side note: we COULD remove all this code, but then the horizon would be black
-
-            Entity entity = camera.getFocusedEntity();
-
-            float f;
-            float g;
-            float h;
-            float r;
-            float s;
-            float t;
-
-
-
-
-            //this is different
-
-            r = 0.25F + 0.75F * (float) viewDistance / 32.0F;
-            r = 1.0F - (float) Math.pow((double) r, 0.25);
-            Vec3d vec3d = world.getSkyColor(camera.getPos(), tickDelta);
-            s = (float) vec3d.x;
-            t = (float) vec3d.y;
-            float u = (float) vec3d.z;
-            float v = MathHelper.clamp(MathHelper.cos(world.getSkyAngle(tickDelta) * 6.2831855F) * 2.0F + 0.5F, 0.0F, 1.0F);
-            BiomeAccess biomeAccess = world.getBiomeAccess();
-            Vec3d vec3d2 = camera.getPos().subtract(2.0, 2.0, 2.0).multiply(0.25);
-            Vec3d vec3d3 = CubicSampler.sampleColor(vec3d2, (x, y, z) -> {
-                return world.getDimensionEffects().adjustFogColor(Vec3d.unpackRgb(((Biome) biomeAccess.getBiomeForNoiseGen(x, y, z).value()).getFogColor()), v);
-            });
-            red = (float) vec3d3.getX();
-            green = (float) vec3d3.getY();
-            blue = (float) vec3d3.getZ();
-            if (viewDistance >= 4) {
-                f = MathHelper.sin(world.getSkyAngleRadians(tickDelta)) > 0.0F ? -1.0F : 1.0F;
-                Vector3f vector3f = new Vector3f(f, 0.0F, 0.0F);
-                h = camera.getHorizontalPlane().dot(vector3f);
-                if (h < 0.0F) {
-                    h = 0.0F;
-                }
-
-                if (h > 0.0F) {
-                    float[] fs = world.getDimensionEffects().getFogColorOverride(world.getSkyAngle(tickDelta), tickDelta);
-                    if (fs != null) {
-                        h *= fs[3];
-                        red = red * (1.0F - h) + fs[0] * h;
-                        green = green * (1.0F - h) + fs[1] * h;
-                        blue = blue * (1.0F - h) + fs[2] * h;
-                    }
-                }
-            }
-
-            red += (s - red) * r;
-            green += (t - green) * r;
-            blue += (u - blue) * r;
-
-
+    @Inject(method = "applyFog", at = @At("HEAD"), cancellable = true)
+    private static void applyFog(Camera camera, BackgroundRenderer.FogType fogType, float viewDistance, boolean thickFog, float tickDelta, CallbackInfo ci) {
+        if (!Methane.ModActive) return;
```

Comment:
not so sure if this check needs to be here-  I think the changes might be better if they persist.

Reply:
yes you could do that, but then the "toggle methane" key would only toggle the fullbright functionality, which is not the entire mod

Reply:
I think this calls for a config option in that case.

Reply:
https://github.com/AnOpenSauceDev/Methane-mod/commit/b998c5c435c80f47f61e7bdbf9c037b58677bab5 kinda fixes this issue